Bottom: dcf215f43282ba193897c00d3d5dec159f5f6a37
Top:    869637c8d50d60612883f939db61fbef86733ac0
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-15 13:20:41 +0800

Refresh of virtio-net

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index f573614..2f762f2 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -476,7 +476,7 @@ static struct sk_buff *receive_small(struct net_device *dev,
 err_xdp:
 	rcu_read_unlock();
 	dev->stats.rx_dropped++;
-	put_page(virt_to_head_page(buf));
+	virtnet_rx_recycle(rq, virt_to_head_page(buf));
 xdp_xmit:
 	return NULL;
 }
@@ -714,6 +714,7 @@ static struct sk_buff *receive_mergeable(struct net_device *dev,
 	return head_skb;
 
 err_xdp:
+	/* FIXME drop */
 	rcu_read_unlock();
 err_skb:
 	put_page(page);
